src/app/clusters/tls-client-management-server/tls-client-management-server.cpp

```diff
@@ -0,0 +1,244 @@
+    if (err != CHIP_NO_ERROR)
+    {
+        mMaxProvisioned = storedMaxProvisioned;
+    }
+    else
+    {
+        // otherwise defaults
+        ChipLogDetail(
+            Zcl, "TlsClientManagement: Unable to load the MaxProvisioned attribute from the KVS. Defaulting to %u",
+            mMaxProvisioned);
+    }
+}
+
+// AttributeAccessInterface
+CHIP_ERROR TlsClientManagementServer::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    VerifyOrDie(aPath.mClusterId == TlsClientManagement::Id);
+
+    switch (aPath.mAttributeId)
+    {
+    case MaxProvisioned::Id:
+        return aEncoder.Encode(mMaxProvisioned);
+    case ProvisionedEndpoints::Id:
+        TlsClientManagementServer * server = this;
+        CHIP_ERROR err =
+            aEncoder.EncodeList([server](const auto & encoder) -> CHIP_ERROR { return server->EncodeProvisionedEndpoints(encoder); });
+        return err;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+uint8_t TlsClientManagementServer::GetMaxProvisioned() const
+{
+    return mMaxProvisioned;
+}
+
+// helper method to get the TlsClientManagement Sounds one by one and encode into a list
+CHIP_ERROR
+TlsClientManagementServer::EncodeProvisionedEndpoints(const AttributeValueEncoder::ListEncodeHelper & encoder)
+{
+    for (uint8_t i = 0; true; i++)
+    {
+        TLSEndpointStruct::Type endpoint;
+
+        auto err = mDelegate.GetProvisionedEndpointByIndex(i, endpoint);
+        if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
+        {
+            return CHIP_NO_ERROR;
+        }
+
+        ReturnErrorOnFailure(err);
+
+        ReturnErrorOnFailure(encoder.Encode(endpoint));
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR TlsClientManagementServer::Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder)
+{
+    VerifyOrDie(aPath.mClusterId == TlsClientManagement::Id);
+    Status status;
+
+    switch (aPath.mAttributeId)
+    {
+    case MaxProvisioned::Id: {
+        uint8_t newValue;
+        ReturnErrorOnFailure(aDecoder.Decode(newValue));
+        status = SetMaxProvisioned(newValue);
+        return StatusIB(status).ToChipError();
+    }
+    default:
+        // Unknown attribute
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    }
+}
+
+Status TlsClientManagementServer::SetMaxProvisioned(uint8_t maxProvisioned)
+{
+    bool enableChanged = mMaxProvisioned != maxProvisioned;
+
+    if (enableChanged)
+    {
+        mMaxProvisioned = maxProvisioned;
+
+        // Write new value to persistent storage.
+        auto endpointId            = GetEndpointId();
+        ConcreteAttributePath path = ConcreteAttributePath(endpointId, TlsClientManagement::Id, MaxProvisioned::Id);
+        GetSafeAttributePersistenceProvider()->WriteScalarValue(path, mMaxProvisioned);
+
+        // and mark as dirty
+        MatterReportingAttributeChangeCallback(path);
```

Comment:
This attribute is marked F in the spec, so it can only change if ConfigurationVersion is updated....

In practice, should its value just be a constructor argument for this server class?

Reply:
Done